python/legate_dataframe/experimental_ibis/__init__.py

```diff
@@ -0,0 +1,309 @@
+        currently.
+        """
+        if table_name is None:
+            table_name = _gen_name("read_csv", str(source))
+
+        usecols, dtypes = get_names_dtypes_from_schema(schema)
+
+        # TODO: May need to adjust "source" for some file schemas.
+        table = legate_dataframe.lib.csv.csv_read(
+            source, delimiter=delimiter, usecols=usecols, dtypes=dtypes
+        )
+
+        # We already have the schema, just insert:
+        self._tables[table_name] = (table, schema)
+        return self.table(table_name)
+
+    def read_parquet(
+        self,
+        source: str | pathlib.Path,
+        table_name: str | None = None,
+    ):
+        if table_name is None:
+            table_name = _gen_name("read_parquet", str(source))
+
+        table = legate_dataframe.lib.parquet.parquet_read(source)
+        return self.create_table(table_name, table)
+
+    def list_tables(self, like=None, database=None):
+        return self._filter_with_like(list(self._tables.keys()), like)
+
+    def table(self, name: str, schema: sch.Schema | None = None):
+        inferred_schema = self.get_schema(name)
+        overridden_schema = {**inferred_schema, **(schema or {})}
+        return ops.DatabaseTable(name, overridden_schema, self).to_expr()
+
+    def get_schema(self, table_name, *, database=None):
+        _, schema = self._tables[table_name]
+
+        return schema
+
+    def compile(self, expr, *args, **kwargs):
+        # We don't have a different intermediate representation, so do nothing
+        return expr
+
+    def create_table(
+        self,
+        name: str,
+        obj=None,
+        *,
+        schema: sch.Schema | None = None,
+        database: str | None = None,
+        temp: bool | None = None,
+        overwrite: bool = False,
+    ) -> ir.Table:
+        """Create a table."""
+        if temp is not None and not temp:
+            com.IbisError(
+                "Passing `temp=False` to the Legate backend's `create_table()` method "
+                "is not supported: all tables are in memory and temporary."
+            )
+
+        if database is not None:
+            com.IbisError(
+                "Passing `database` to the Legate backend's `create_table()` method "
+                "is not supported: Legate cannot set a database."
+            )
+
+        if name in self._tables and not overwrite:
+            raise com.IbisError(f"Cannot overwrite existing table `{name}`")
+
+        if obj is None and schema is None:
+            raise com.IbisError("The schema or obj parameter is required")
+
+        if schema is not None:
+            schema = ibis.schema(schema)
+
+        if obj is not None:
+            table = self._convert_object(obj)
+            if schema is None:
+                schema = infer_schema_from_logical_table(table)
+            else:
+                # Check if schema matches actual table (is it right to not
+                # enforce the schema by dropping cols or casting?)
```

Comment:
We try not to infer intent because it's a tricky problem.

1. w.r.t. column names, we definitely don't, because it's easy for the "wrong" schema to match a subset of columns and cause problems.  If users want to drop columns, this is what `mutate` and `select` are for.
2. w.r.t dtypes, this if more doable and we have a `castable` utility for determining lowest-common-denominator dtypes, but these decisions aren't always perfectly deterministic.

Reply:
OK, I read that as this check making sense and will remove the comment, easy to soften the error eventually (I am not sure the pandas/polars backend were/are strict about it, so the question).